### ISSUE TITLE
Machine status marked as 'Timeout' after host suspend ended with Virtualbox

### ISSUE THREAD

**Issue by smalot** [docker/machine#3211]
Hi all,

Using the `Virtualbox` driver, if machines are running when I suspend my computer (laptop), machines are still running after wakeup in `Virtualbox` GUI (in command line too), but `docker-machine` can't perform to connect to them and marked VMs as `Timeout`.

However, `docker-machine` is able to connect using ssh.

Steps to reproduce:

````
docker-machine rm -f default
docker-machine create --driver virtualbox default
docker-machine ls
# -> the new machine is correctly marked as running
# action: suspend host computer
# action: wakeup host computer
docker-machine ls
# -> the new machine is marked as timeout
docker-machine ssh default
# -> the ssh command works and we are connected to the default machine
````

Here's detail about my global setup:

````sh
~$ lsb_release  -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu Xenial Xerus (development branch)
Release:	16.04
Codename:	xenial

~$ docker-machine -v
docker-machine version 0.6.0, build e27fb87

~$ VBoxManage --version
5.0.16_Ubuntur105871
````

If you need more details, I'll be happy to help you.

Many thanks,

**Comment by smalot** [comment]
Important thing, the following command line indicate the correct state of the VM:

````sh
docker-machine status default
````

So the issue seems to be limited to the "docker-machine ls" command.

**Comment by nathanleclaire** [comment]
What is their status listed as with `docker-machine status`?

**Comment by smalot** [comment]
With `docker-machine status` the result is `Running` which is the good one.

**Comment by nathanleclaire** [comment]
Can you post the output of `ls` with `docker-machine --debug` enabled please?

**Comment by smalot** [comment]
Here is the result of `docker-machine --debug ls` command:

````
Docker Machine Version:  0.6.0, build e27fb87
ls timeout set to 10s
Found binary path at /usr/local/bin/docker-machine
Launching plugin server for driver virtualbox
Plugin server listening at address 127.0.0.1:40113
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(ava) Calling .GetURL
(ava) DBG | COMMAND: /usr/bin/VBoxManage showvminfo ava --machinereadable
(ava) DBG | STDOUT:
(ava) DBG | {
(ava) DBG | name="ava"
(ava) DBG | groups="/"
(ava) DBG | ostype="Linux 2.6 / 3.x / 4.x (64-bit)"
(ava) DBG | UUID="44a96b2b-f0c7-4253-b581-4b5040323181"
(ava) DBG | CfgFile="/home/sebastien/.docker/machine/machines/ava/ava/ava.vbox"
(ava) DBG | SnapFldr="/home/sebastien/.docker/machine/machines/ava/ava/Snapshots"
(ava) DBG | LogFldr="/home/sebastien/.docker/machine/machines/ava/ava/Logs"
(ava) DBG | hardwareuuid="44a96b2b-f0c7-4253-b581-4b5040323181"
(ava) DBG | memory=1024
(ava) DBG | pagefusion="off"
(ava) DBG | vram=8
(ava) DBG | cpuexecutioncap=100
(ava) DBG | hpet="on"
(ava) DBG | chipset="piix3"
(ava) DBG | firmware="BIOS"
(ava) DBG | cpus=1
(ava) DBG | pae="on"
(ava) DBG | longmode="on"
(ava) DBG | cpuid-portability-level=0
(ava) DBG | bootmenu="disabled"
(ava) DBG | boot1="dvd"
(ava) DBG | boot2="dvd"
(ava) DBG | boot3="disk"
(ava) DBG | boot4="none"
(ava) DBG | acpi="on"
(ava) DBG | ioapic="on"
(ava) DBG | biossystemtimeoffset=0
(ava) DBG | rtcuseutc="on"
(ava) DBG | hwvirtex="on"
(ava) DBG | nestedpaging="on"
(ava) DBG | largepages="on"
(ava) DBG | vtxvpid="on"
(ava) DBG | vtxux="on"
(ava) DBG | paravirtprovider="default"
(ava) DBG | VMState="running"
(ava) DBG | VMStateChangeTime="2016-03-22T22:34:45.361000000"
(ava) DBG | monitorcount=1
(ava) DBG | accelerate3d="off"
(ava) DBG | accelerate2dvideo="off"
(ava) DBG | teleporterenabled="off"
(ava) DBG | teleporterport=0
(ava) DBG | teleporteraddress=""
(ava) DBG | teleporterpassword=""
(ava) DBG | tracing-enabled="off"
(ava) DBG | tracing-allow-vm-access="off"
(ava) DBG | tracing-config=""
(ava) DBG | autostart-enabled="off"
(ava) DBG | autostart-delay=0
(ava) DBG | defaultfrontend=""
(ava) DBG | storagecontrollername0="SATA"
(ava) DBG | storagecontrollertype0="IntelAhci"
(ava) DBG | storagecontrollerinstance0="0"
(ava) DBG | storagecontrollermaxportcount0="30"
(ava) DBG | storagecontrollerportcount0="30"
(ava) DBG | storagecontrollerbootable0="on"
(ava) DBG | "SATA-0-0"="/home/sebastien/.docker/machine/machines/ava/boot2docker.iso"
(ava) DBG | "SATA-ImageUUID-0-0"="da4f3358-e9f5-488c-9929-769f3e6a198c"
(ava) DBG | "SATA-tempeject"="off"
(ava) DBG | "SATA-IsEjected"="off"
(ava) DBG | "SATA-1-0"="/home/sebastien/.docker/machine/machines/ava/disk.vmdk"
(ava) DBG | "SATA-ImageUUID-1-0"="95a714d2-9e48-4597-b676-3efc78daa916"
(ava) DBG | "SATA-2-0"="none"
(ava) DBG | "SATA-3-0"="none"
(ava) DBG | "SATA-4-0"="none"
(ava) DBG | "SATA-5-0"="none"
(ava) DBG | "SATA-6-0"="none"
(ava) DBG | "SATA-7-0"="none"
(ava) DBG | "SATA-8-0"="none"
(ava) DBG | "SATA-9-0"="none"
(ava) DBG | "SATA-10-0"="none"
(ava) DBG | "SATA-11-0"="none"
(ava) DBG | "SATA-12-0"="none"
(ava) DBG | "SATA-13-0"="none"
(ava) DBG | "SATA-14-0"="none"
(ava) DBG | "SATA-15-0"="none"
(ava) DBG | "SATA-16-0"="none"
(ava) DBG | "SATA-17-0"="none"
(ava) DBG | "SATA-18-0"="none"
(ava) DBG | "SATA-19-0"="none"
(ava) DBG | "SATA-20-0"="none"
(ava) DBG | "SATA-21-0"="none"
(ava) DBG | "SATA-22-0"="none"
(ava) DBG | "SATA-23-0"="none"
(ava) DBG | "SATA-24-0"="none"
(ava) DBG | "SATA-25-0"="none"
(ava) DBG | "SATA-26-0"="none"
(ava) DBG | "SATA-27-0"="none"
(ava) DBG | "SATA-28-0"="none"
(ava) DBG | "SATA-29-0"="none"
(ava) DBG | natnet1="nat"
(ava) DBG | macaddress1="080027DB1F15"
(ava) DBG | cableconnected1="on"
(ava) DBG | nic1="nat"
(ava) DBG | nictype1="82540EM"
(ava) DBG | nicspeed1="0"
(ava) DBG | mtu="0"
(ava) DBG | sockSnd="64"
(ava) DBG | sockRcv="64"
(ava) DBG | tcpWndSnd="64"
(ava) DBG | tcpWndRcv="64"
(ava) DBG | Forwarding(0)="ssh,tcp,127.0.0.1,38367,,22"
(ava) DBG | hostonlyadapter2="vboxnet1"
(ava) DBG | macaddress2="080027F7362C"
(ava) DBG | cableconnected2="on"
(ava) DBG | nic2="hostonly"
(ava) DBG | nictype2="82540EM"
(ava) DBG | nicspeed2="0"
(ava) DBG | nic3="none"
(ava) DBG | nic4="none"
(ava) DBG | nic5="none"
(ava) DBG | nic6="none"
(ava) DBG | nic7="none"
(ava) DBG | nic8="none"
(ava) DBG | hidpointing="ps2mouse"
(ava) DBG | hidkeyboard="ps2kbd"
(ava) DBG | uart1="off"
(ava) DBG | uart2="off"
(ava) DBG | uart3="off"
(ava) DBG | uart4="off"
(ava) DBG | lpt1="off"
(ava) DBG | lpt2="off"
(ava) DBG | audio="none"
(ava) DBG | clipboard="disabled"
(ava) DBG | draganddrop="disabled"
(ava) DBG | SessionName="headless"
(ava) DBG | VideoMode="720,400,0"@0,0 1
(ava) DBG | vrde="off"
(ava) DBG | usb="off"
(ava) DBG | ehci="off"
(ava) DBG | xhci="off"
(ava) DBG | SharedFolderNameMachineMapping1="hosthome"
(ava) DBG | SharedFolderPathMachineMapping1="/home"
(ava) DBG | VRDEActiveConnection="off"
(ava) DBG | VRDEClients=0
(ava) DBG | vcpenabled="off"
(ava) DBG | vcpscreens=0
(ava) DBG | vcpfile="/home/sebastien/.docker/machine/machines/ava/ava/ava.webm"
(ava) DBG | vcpwidth=1024
(ava) DBG | vcpheight=768
(ava) DBG | vcprate=512
(ava) DBG | vcpfps=25
(ava) DBG | GuestMemoryBalloon=0
(ava) DBG | GuestOSType="Linux26_64"
(ava) DBG | GuestAdditionsRunLevel=2
(ava) DBG | GuestAdditionsVersion="5.0.16 r105871"
(ava) DBG | GuestAdditionsFacility_VirtualBox Base Driver=50,1458393944372
(ava) DBG | GuestAdditionsFacility_VirtualBox System Service=50,1458393952315
(ava) DBG | GuestAdditionsFacility_Seamless Mode=0,1458686050058
(ava) DBG | GuestAdditionsFacility_Graphics Mode=0,1458686050058
(ava) DBG | }
(ava) DBG | STDERR:
(ava) DBG | {
(ava) DBG | }
(ava) DBG | Using SSH client type: external
(ava) DBG | {[-o BatchMode=yes -o PasswordAuthentication=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none docker@127.0.0.1 -o IdentitiesOnly=yes -i /home/sebastien/.docker/machine/machines/ava/id_rsa -p 38367] /usr/bin/ssh}
(ava) DBG | About to run SSH command:
(ava) DBG | ip addr show dev eth1
(ava) DBG | SSH cmd err, output: <nil>: 4: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
(ava) DBG |     link/ether 08:00:27:f7:36:2c brd ff:ff:ff:ff:ff:ff
(ava) DBG |     inet 192.168.99.102/24 brd 192.168.99.255 scope global eth1
(ava) DBG |        valid_lft forever preferred_lft forever
(ava) DBG |     inet6 fe80::a00:27ff:fef7:362c/64 scope link 
(ava) DBG |        valid_lft forever preferred_lft forever
(ava) DBG | 
Reading CA certificate from /home/sebastien/.docker/machine/certs/ca.pem
(ava) DBG | SSH returned: 4: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
(ava) DBG |     link/ether 08:00:27:f7:36:2c brd ff:ff:ff:ff:ff:ff
(ava) DBG |     inet 192.168.99.102/24 brd 192.168.99.255 scope global eth1
(ava) DBG |        valid_lft forever preferred_lft forever
Reading server certificate from /home/sebastien/.docker/machine/machines/ava/server.pem
(ava) DBG |     inet6 fe80::a00:27ff:fef7:362c/64 scope link 
(ava) DBG |        valid_lft forever preferred_lft forever
(ava) DBG | 
Reading server key from /home/sebastien/.docker/machine/machines/ava/server-key.pem
(ava) DBG | END SSH
(ava) DBG | 
(ava) Calling .DriverName
NAME   ACTIVE   DRIVER       STATE     URL   SWARM   DOCKER   ERRORS
ava             virtualbox   Timeout                          
Making call to close driver server
(ava) Calling .Close
Successfully made call to close driver server
Making call to close connection to plugin binary
(ava) DBG | Closing plugin on server side
````

**Comment by nathanleclaire** [comment]
Hm, all the calls seem to be returning fairly normally.  I wonder what's going on.

**Comment by smalot** [comment]
That's quite strange. Previously I encountered just issue on `docker-machine ls` which indicate a `Timeout` on status.
Now VMs are flagged as `Running` like that:

````sh
NAME   ACTIVE   DRIVER       STATE     URL                         SWARM   DOCKER    ERRORS
ava    -        virtualbox   Running   tcp://192.168.99.102:2376           Unknown   Unable to query docker version: Get https://192.168.99.102:2376/v1.15/version: dial tcp 192.168.99.102:2376: getsockopt: no route to host
````

After few checks, it appears that Virtualbox doesn't appreciate the suspend mode, and cut the `vboxnet1` network interface, or at least, forgot to enable again it after resume.
See: http://askubuntu.com/questions/626741/ubuntu-15-04-disconnects-from-virtual-networks-after-wakeup

**Comment by saikojosh** [comment]
**+1**

I have the same issue and it's permanent, even rebooting the host machine doesn't work. The only thing that works for me is destroying the docker machine and re-creating it.

My SO question: http://stackoverflow.com/questions/37069718/docker-machine-timeout-how-to-fix-without-destroying-the-machine

**Comment by davidraleigh** [comment]
I added a new machine with the google driver google

```bash
docker-machine create --driver google --google-project project-name --google-zone us-central1-c raleigh-docker-remote-debug
```

And after that I can't connect to any of my vms through `docker-machine`. They all list `Timeout` as their status. Looking at the debug it seems to all go haywire at the `ip addr show` command similar to @smalot.

```
(default) DBG | About to run SSH command:
(default) DBG | ip addr show
```

Maybe the `docker-machine create` call wasn't completed and now my keys are in a broken state?

**Comment by myuseringithub** [comment]
Same here with google driver after a while using it (maybe ip changed or restarted remote vm), it shows status timeout ! cannot fix it.

**Comment by lexicondevil** [comment]
This is an issue I ran into as well, but found a fix in a previously reported issue: https://github.com/docker/machine/issues/1684 

tl;dr: the fix for me was modifying `/etc/hosts` to comment out `::1 localhost` and you should be able to get those VMs functional again!

**Comment by nathanleclaire** [comment]
@lexicondevil Was this with a recent version of Machine? For quite a while now we've attempted to refer to `127.0.0.1` specifically to avoid that problem.

**Comment by lexicondevil** [comment]
I'm on 0.8.2, so I might be out of the loop on any fix that's been released. My apologies for any out of date information on my part!

**Comment by myuseringithub** [comment]
For me the issue was concerning 'docker-user' default user used for ssh`ing which was different that my own. https://github.com/docker/machine/issues/3522#issuecomment-280275707
Now each time I create a docker-machine I always make a backup for the .docker/machines, just in case, would save me a lot of time.

**Comment by supermock** [comment]
In my case I solved this problem by checking if the network interface set in the virtual machine was active on my host with `sudo ifconfig`, well... as expected was not enabled, so I just ran `sudo ifconfig vboxnet0 up`, and no more timeout in `docker-machine ls`.

**Comment by NorseGaud** [comment]
My issue was right after a reboot, it was trying to start and then timing out. Must have been trying to start too fast since I can manually try again and it works. I had to create a separate .plist to get it to work:

```
cat > ~/Library/LaunchAgents/cicd.start-docker-machine.plist <<EOD
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
    <dict>
        <key>EnvironmentVariables</key>
        <dict>
            <key>PATH</key>
            <string>/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin</string>
        </dict>
        <key>Label</key>
        <string>cicd.start-docker-machine</string>
        <key>ProgramArguments</key>
        <array>
            <string>/usr/local/bin/docker-machine</string>
            <string>start</string>
            <string>default</string>
        </array>
        <key>UserName</key>
        <string>cicd</string>
        <key>RunAtLoad</key>
        <true/>
    </dict>
</plist>
```
